rocketpy/simulation/multivariate_rejection_sampler.py

```diff
@@ -0,0 +1,259 @@
+
+
+class MultivariateRejectionSampler:
+    """Class that performs Multivariate Rejection Sampling (MRS) from MonteCarlo
+    results. The class currently assumes that all input variables are independent
+    when performing the
+
+    Attributes
+    ----------
+    """
+
+    def __init__(
+        self,
+        monte_carlo_filepath,
+        mrs_filepath,
+    ):
+        """Initializes Multivariate Rejection Sampler (MRS) class
+
+        Parameters
+        ----------
+        monte_carlo_filepath : str
+            Filepath prefixes to the files created from a MonteCarlo simulation
+            results and used as input for resampling.
+        mrs_filepath : str
+            Filepath prefix to MRS obtained samples. The files created follow the same
+            structure as those created by the MonteCarlo class but now containing the
+            selected sub-samples.
+
+        Returns
+        -------
+        None
+        """
+        self.monte_carlo_filepath = monte_carlo_filepath
+        self.mrs_filepath = mrs_filepath
+        self.distribution_dict = None
+        self.original_sample_size = 0
+        self.sup_ratio = 1
+        self.expected_sample_size = None
+        self.final_sample_size = None
+        self.input_variables_names = set()
+        self.output_variables_names = set()
+        self.all_sample_list = []
+        self.accepted_sample_list = []
+        self.__setup_input()
+        self.__load_output()
+
+    def __setup_input(self):
+        """Loads input information from monte carlo in a SampleInformation
+        object
+        """
+        input_filename = f"{self.monte_carlo_filepath}.inputs.txt"
+
+        try:
+            input_file = open(input_filename, "r+", encoding="utf-8")
+        except FileNotFoundError as e:
+            raise FileNotFoundError(
+                f"Input file from monte carlo {input_filename} not found!"
+            ) from e
+
+        try:
+            for line in input_file.readlines():
+                sample_info = SampleInformation()
+                line_json = json.loads(line)
+                sample_info.inputs_json = line_json
+                self.all_sample_list.append(sample_info)
+
+                # sets and validates input variables names
+                if not self.input_variables_names:
+                    self.input_variables_names = set(line_json.keys())
+                else:
+                    if self.input_variables_names != set(line_json.keys()):
+                        raise ValueError(
+                            "Input file from Monte Carlo contains different "
+                            "variables for different lines"
+                        )
+                self.original_sample_size += 1
+        except Exception as e:
+            raise ValueError(
+                "An error occurred while reading "
+                f"the monte carlo input file {input_filename}!"
+            ) from e
+
+        input_file.close()
```

Comment:
This line should be inside a `finally` block, I believe. The same applies to the method ``load_output``.